v2/api/servicebus/v1beta20210101preview/namespaces_topics_subscription_types_gen.go

```diff
@@ -0,0 +1,1491 @@
+	// +kubebuilder:validation:MinLength=1
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	AzureName string `json:"azureName,omitempty"`
+
+	// DeadLetteringOnFilterEvaluationExceptions: Value that indicates whether a subscription has dead letter support on filter
+	// evaluation exceptions.
+	DeadLetteringOnFilterEvaluationExceptions *bool `json:"deadLetteringOnFilterEvaluationExceptions,omitempty"`
+
+	// DeadLetteringOnMessageExpiration: Value that indicates whether a subscription has dead letter support when a message
+	// expires.
+	DeadLetteringOnMessageExpiration *bool `json:"deadLetteringOnMessageExpiration,omitempty"`
+
+	// DefaultMessageTimeToLive: ISO 8061 Default message timespan to live value. This is the duration after which the message
+	// expires, starting from when the message is sent to Service Bus. This is the default value used when TimeToLive is not
+	// set on a message itself.
+	DefaultMessageTimeToLive *string `json:"defaultMessageTimeToLive,omitempty"`
+
+	// DuplicateDetectionHistoryTimeWindow: ISO 8601 timeSpan structure that defines the duration of the duplicate detection
+	// history. The default value is 10 minutes.
+	DuplicateDetectionHistoryTimeWindow *string `json:"duplicateDetectionHistoryTimeWindow,omitempty"`
+
+	// EnableBatchedOperations: Value that indicates whether server-side batched operations are enabled.
+	EnableBatchedOperations *bool `json:"enableBatchedOperations,omitempty"`
+
+	// ForwardDeadLetteredMessagesTo: Queue/Topic name to forward the Dead Letter message
+	ForwardDeadLetteredMessagesTo *string `json:"forwardDeadLetteredMessagesTo,omitempty"`
+
+	// ForwardTo: Queue/Topic name to forward the messages
+	ForwardTo *string `json:"forwardTo,omitempty"`
+
+	// Location: Location to deploy resource to
+	Location *string `json:"location,omitempty"`
+
+	// LockDuration: ISO 8061 lock duration timespan for the subscription. The default value is 1 minute.
+	LockDuration *string `json:"lockDuration,omitempty"`
+
+	// MaxDeliveryCount: Number of maximum deliveries.
+	MaxDeliveryCount *int `json:"maxDeliveryCount,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// Owner: The owner of the resource. The owner controls where the resource goes when it is deployed. The owner also
+	// controls the resources lifecycle. When the owner is deleted the resource will also be deleted. Owner is expected to be a
+	// reference to a servicebus.azure.com/NamespacesTopic resource
+	Owner *genruntime.KnownResourceReference `group:"servicebus.azure.com" json:"owner,omitempty" kind:"NamespacesTopic"`
+
+	// RequiresSession: Value indicating if a subscription supports the concept of sessions.
+	RequiresSession *bool `json:"requiresSession,omitempty"`
+
+	// Status: Enumerates the possible values for the status of a messaging entity.
+	Status *SBSubscriptionProperties_Status `json:"status,omitempty"`
```

Comment:
I'm wondering if this field should actually be marked read-only and not show up here in the spec?

Reply:
Maybe try setting it either via an ARM template or in one of our tests and see if you get rejected? If so we should maybe prune it from the type in our config and then send them a PR to update their Swagger to mark it as a non-settable property.